test/task_cancel_test.cpp

```diff
@@ -51,11 +51,33 @@ task<int> bar() {
   co_return -1;
 }
 
+task<inplace_stop_token> get_token_inner() {
+  co_return co_await get_stop_token;
+}
+
+task<inplace_stop_token> get_token_outer() {
+  auto a = co_await get_stop_token;
+  auto b = co_await get_token_inner();
+  EXPECT_EQ(a, b);
+  co_return b;
+}
+
 task<void> void_test() {
   co_await stop;
   co_return;
 }
 
+bool didNotStopWhenNotAskedTo = false;
+
+task<int> test_stop_if_requested(inplace_stop_source& stopSource) {
+  co_await stop_if_requested; // shouldn't stop
+  didNotStopWhenNotAskedTo = true;
```

Comment:
The double negatives didn't leave me not confused.

How about `continuedWhenStopWasNotYetRequested`?